packages/gestalt/src/Toast.js

```diff
@@ -40,15 +85,14 @@ type Props = {|
   /**
    * An optional thumbnail image to displayed next to the text.
    */
-  thumbnail?: Node,
+  thumbnail?:
+    | {| image: Element<typeof Image> |}
+    | {| avatar: Element<typeof Avatar> |}
+    | {| icon: Element<typeof Icon> |},
   /**
-   * The masked shape of the thumbnail.
+   * See the [type variant](https://gestalt.pinterest.systems/web/toast#Type) to learn more.
    */
-  thumbnailShape?: 'circle' | 'square',
-  /**
-   * Use the `'error'` variant to indicate an error message. Generally not recommended given the ephemeral nature of Toasts.
-   */
-  variant?: 'default' | 'error',
+  type?: 'success' | 'error' | 'progress',
```

Comment:
I think this needs a `default` to indicate the default state rather than relying on the absence of the prop (or passing a nullish value)